### DISCUSSION TITLE
New Features that Musescore 4 NEED!!!

### DISCUSSION THREAD

**Discussion by Antonio2007Dev** [musescore/MuseScore#12677] [Ideas]
Hi, I was wondering if the following functionalities can be done in musescore 3: or if they can be added for musescore 4:
I don't know the exact name of the functionality in question so I'll give a little explanation of what it is and I'll add some images.

- Play rehearsal marks (ie: be able to play the part that is enclosed by marks A and B, and thus save space in certain parts of the song where it repeats)
![Sendas A](https://user-images.githubusercontent.com/110574403/182734555-20f72e8e-415c-462b-8d29-dafbe3369d1f.jpg)
![Sendas B](https://user-images.githubusercontent.com/110574403/182734572-75e6e680-52de-4fbb-a4b0-7e0f1dd0cce8.jpg)
![Sendas from A to B](https://user-images.githubusercontent.com/110574403/182734580-c019419f-2dad-42cc-b367-c62470ece033.jpg)
Score Link: http://soymusicaecuador.blogspot.com/2010/04/nos-piden-el-pasillo-sendas-distintas.html

- Grace note chords, grace note rests, etc.
![Different grace notes in the same principal note](https://user-images.githubusercontent.com/110574403/182736150-1a829e2e-6173-4f79-aa4b-3fee4376c596.png)
![Grace note rests](https://user-images.githubusercontent.com/110574403/182736153-5ac18c29-35c0-4090-950b-f363e348733c.png)
![2022-08-03](https://user-images.githubusercontent.com/110574403/182736506-a15054e1-bf6a-4148-b6d7-03bdb2f12a52.png)
![More grace notes features](https://user-images.githubusercontent.com/110574403/182736699-5384a788-fbde-4bf0-b5fa-815d4bf70d8e.png)
Score link: http://www.teddyleongshe.com/Hijo%20de%20la%20luna.pdf

- I took a look at the last nocturne update on 08/03/2022, and there are a few things that I dislike a bit: The blue playhead that moves when you touch a score, in musescore 3 it was a thicker bar that wrapped around the entire the note and for my taste was easier to understand, but now in musescore 4 that bar in fine and thin,
and the notes are not painted in the color of the layer... (it would be good if they added the option of having those two bars in preferences, because some people will like one more and others will prefer the other)
![Play bar in musescore 3](https://user-images.githubusercontent.com/110574403/182737572-97c8295e-78ce-45ed-a84c-6153f3cdfa71.png)
![Play bar in musescore 4](https://user-images.githubusercontent.com/110574403/182737574-6f51e278-6a24-476e-ac10-39b786200d70.png)

Thank you for understanding

**Comment by Antonio2007Dev** [comment]
Emm, the 6th picture don´t see please... this image should not go there

**Comment by cbjeukendrup** [comment]
About point 2: I could relatively easily recreate these examples in MuseScore 3.
<img width="776" alt="Schermafbeelding 2022-08-04 om 02 45 43" src="https://user-images.githubusercontent.com/48658420/182739198-b72a2feb-18ff-4022-a405-4359ccaaf28d.png">
[Excerpt.mscz.zip](https://github.com/musescore/MuseScore/files/9257764/Excerpt.mscz.zip)

Use the following features: 
- make sure to select the "advanced" workspace using the dropdown in the top right corner of the window
- use multiple "voices" (the 1 2 3 4 buttons in the note input toolbar)
- enter the main melody in voice 1, and then other notes that should have the stem downwards in voice 2 and notes with stem upwards in voice 3
- In voice 2, 3 and 4, you can select rests and press delete to hide them
- you can press the letter X to flip a note or other element (change stem direction / slur direction)
- for the rests under the beams, use normal rests, and then in the "Beam properties" palette, select the "Start beam" option
- in the Inspector, use the "small" option for chords and rests to make them small like in the example.

(I see no "grace rests" in your examples though; those are just normal rests a different voice, with the beaming option and the small option enabled.)

Things like slurs and beams have been massively improved in MuseScore 4, so most probably this will look _much_ better in 4.

**Comment by ghost** [comment]
(Yes, but please nobody look at it in MU4 until we fix the beams starting on rests) 😇 
Update: see #12802
